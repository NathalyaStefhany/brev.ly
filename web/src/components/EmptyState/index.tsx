import React, { ComponentProps, ReactElement } from 'react';

type EmptyStateProps = {
  icon: ReactElement;
  description: string;
} & ComponentProps<'div'>;

export const EmptyState: React.FC<EmptyStateProps> = ({
  icon,
  description,
  ...props
}) => {
  return (
    <div
      className="flex flex-col items-center justify-center gap-6 mt-16 mb-12"
      {...props}
    >
      {icon}

      <p className="text-xs text-gray-500 uppercase text-center">
        {description}
      </p>
    </div>
  );
};
