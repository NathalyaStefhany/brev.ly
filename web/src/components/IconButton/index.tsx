import React, { ComponentProps, ComponentType } from 'react';

import { IconProps } from '@phosphor-icons/react';

type IconButtonProps = ComponentProps<'button'> & {
  icon: ComponentType<IconProps>;
};

export const IconButton: React.FC<IconButtonProps> = ({
  icon: Icon,
  ...props
}) => {
  return (
    <button
      type="button"
      className="cursor-pointer transition-all duration-300 bg-gray-200 border border-gray-200 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50 disabled:cursor-not-allowed"
      {...props}
    >
      <Icon size="1rem" color="var(--color-gray-600)" />
    </button>
  );
};
