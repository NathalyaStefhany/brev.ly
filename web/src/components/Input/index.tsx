import { Warning } from '@phosphor-icons/react';
import React, { ComponentProps } from 'react';

import { tv } from 'tailwind-variants';

const labelVariants = tv({
  base: 'text-xs text-gray-500 uppercase focus:font-bold peer-focus:font-bold',
  variants: {
    intent: {
      default: 'peer-focus:text-blue-base',
      error: 'peer-focus:text-danger',
    },
  },
  defaultVariants: {
    intent: 'default',
  },
});

const inputVariants = tv({
  base: 'peer text-md text-gray-600 font-normal border border-gray-300 caret-blue-base rounded-lg px-8 py-7 placeholder:text-gray-400',
  variants: {
    intent: {
      default: 'focus:outline-blue-base',
      error: 'focus:outline-danger',
    },
  },
  defaultVariants: {
    intent: 'default',
  },
});

type InputProps = ComponentProps<'input'> & {
  id: string;
  label: string;
  error?: string;
};

export const Input: React.FC<InputProps> = ({ id, label, error, ...props }) => {
  const intent = error ? 'error' : 'default';

  return (
    <div className="max-w-176 w-full flex flex-col-reverse gap-4">
      {error && (
        <div className="flex flex-row items-center gap-4">
          <Warning size="1rem" color="var(--color-danger)" />

          <span
            id={`${id}-error`}
            className="text-sm text-gray-500"
            data-testid={`${id}-error`}
          >
            {error}
          </span>
        </div>
      )}

      <input
        id={id}
        type="text"
        aria-invalid={!!error}
        aria-describedby={`${id}-error`}
        className={inputVariants({ intent })}
        {...props}
      />

      <label htmlFor={id} className={labelVariants({ intent })}>
        {label}
      </label>
    </div>
  );
};
