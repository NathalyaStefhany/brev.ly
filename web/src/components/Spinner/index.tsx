import React from 'react';

export const Spinner: React.FC = () => {
  return (
    <div className="relative w-12 h-12">
      {[...Array(10)].map((_, i) => (
        <div
          // eslint-disable-next-line react/no-array-index-key
          key={i}
          className="absolute top-4.5 left-5.5 w-1 h-3 bg-blue-base rounded origin-center animate-fade"
          style={{
            transform: `rotate(${i * 36}deg) translateY(-150%)`,
            opacity: i / 10,
            animationDelay: `${i * 0.098}s`,
          }}
        />
      ))}
    </div>
  );
};
