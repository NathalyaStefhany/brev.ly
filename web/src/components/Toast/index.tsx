import React from 'react';

import * as ToastRadix from '@radix-ui/react-toast';
import { WarningCircle, X } from '@phosphor-icons/react';

type ToastProps = {
  id: string;
  type: 'error' | 'information';
  title: string;
  description: string;
  duration?: number;
  open: boolean;
  onOpenChange: (open: boolean) => void;
};

export const Toast: React.FC<ToastProps> = ({
  id,
  type,
  title,
  description,
  duration = 5000,
  open,
  onOpenChange,
}) => {
  const color = type === 'error' ? 'danger' : 'blue-base';

  const textColor = `text-${color}`;
  const backgroundColor = type === 'error' ? '#f1d4da' : '#d6d8ef';

  return (
    <ToastRadix.Provider swipeDirection="right">
      <ToastRadix.Root
        open={open}
        onOpenChange={onOpenChange}
        duration={duration}
        className={`max-w-screen w-180 flex flex-row gap-6 items-start justify-start bg-[${backgroundColor}] p-8 rounded-lg shadow-lg shadow-gray-300`}
        data-testid={id}
      >
        <WarningCircle
          color={`var(--color-${color})`}
          size="1.25rem"
          weight="fill"
        />

        <div className="flex-1">
          <ToastRadix.Title className={`text-md ${textColor}`}>
            {title}
          </ToastRadix.Title>

          <ToastRadix.Description className={`text-sm ${textColor} mt-2`}>
            {description}
          </ToastRadix.Description>
        </div>

        <ToastRadix.Action
          altText="BotÃ£o para fechar a caixa de aviso"
          className={`cursor-pointer ${textColor}`}
        >
          <X />
        </ToastRadix.Action>
      </ToastRadix.Root>

      <ToastRadix.Viewport className="z-3 fixed bottom-6 left-1/2 transform -translate-x-1/2 md:bottom-12 md:right-12 md:left-auto md:transform-none md:translate-x-0" />
    </ToastRadix.Provider>
  );
};
