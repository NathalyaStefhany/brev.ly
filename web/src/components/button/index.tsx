import React, { ComponentProps, ComponentType } from 'react';

import { tv, VariantProps } from 'tailwind-variants';
import { IconProps } from '@phosphor-icons/react';

const buttonVariants = tv({
  base: 'flex flex-row gap-3 items-center justify-center cursor-pointer transition-all duration-300 disabled:cursor-not-allowed',
  variants: {
    variant: {
      primary:
        'max-w-176 w-full bg-blue-base text-md text-white rounded-lg px-10 py-7.5 hover:not-disabled:bg-blue-dark disabled:opacity-50',
      secondary:
        'bg-gray-200 border border-gray-200 text-sm font-semibold text-gray-500 rounded-sm p-3.5 hover:not-disabled:border-blue-base disabled:opacity-50',
    },
  },
  defaultVariants: {
    variant: 'primary',
  },
});

type ButtonProps = ComponentProps<'button'> &
  VariantProps<typeof buttonVariants> & {
    icon?: ComponentType<IconProps>;
  };

export const Button: React.FC<ButtonProps> = ({
  variant,
  icon: Icon,
  children,
  ...props
}) => {
  return (
    <button type="button" className={buttonVariants({ variant })} {...props}>
      {Icon && (
        <Icon
          size="1rem"
          color="var(--color-gray-600)"
          data-testid="button-icon"
        />
      )}

      {children}
    </button>
  );
};
