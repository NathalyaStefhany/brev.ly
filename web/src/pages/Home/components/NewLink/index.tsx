import React, { useState } from 'react';

import { useForm } from 'react-hook-form';
import * as Toast from '@radix-ui/react-toast';
import { zodResolver } from '@hookform/resolvers/zod';
import { z } from 'zod';
import { Input } from '@/components/Input';
import { Button } from '@/components/Button';
import { api } from '@/service/api';
import { WarningCircle, X } from '@phosphor-icons/react';
import { AxiosError } from 'axios';
import { queryClient } from '@/service/queryClient';

const originalLinkErrorMessage = 'Informe uma url válida.';
const shortenedLinkErrorMessage =
  'Informe uma url minúscula e sem espaço/caracter especial.';

const newLinkSchema = z
  .object({
    originalLink: z
      .string()
      .min(1, originalLinkErrorMessage)
      .refine((value) => {
        try {
          new URL(value);

          return true;
        } catch {
          return /^(www\.)[\w-]+\.[a-z]{2,}(\.[a-z]{2,})?(\/[^\s]*)?$/i.test(
            value,
          );
        }
      }, originalLinkErrorMessage),
    shortenedLink: z
      .string()
      .min(1, shortenedLinkErrorMessage)
      .regex(/^[a-z0-9-]+$/, shortenedLinkErrorMessage),
  })
  .required();

type NewLinkData = z.infer<typeof newLinkSchema>;

type ErrorData = {
  message?: string;
};

export const NewLink: React.FC = () => {
  const [openToastError, setOpenToastError] = useState(false);
  const [errorCreatingNewLink, setErrorCreatingNewLink] = useState({
    title: '',
    description: '',
  });

  const {
    register,
    handleSubmit,
    formState: { errors, isSubmitting },
    reset,
  } = useForm({
    resolver: zodResolver(newLinkSchema),
  });

  const handleSaveLink = async (data: NewLinkData): Promise<void> => {
    try {
      await api.post('/shortened-links', data);

      queryClient.refetchQueries({ queryKey: ['links list'], exact: true });

      reset();
    } catch (error) {
      const err = error as AxiosError;

      let description = 'Por favor, tente novamente mais tarde.';

      const errorData = err.response?.data as ErrorData;

      if (
        err.status === 400 &&
        errorData?.message === 'Shortened link already exists'
      ) {
        description = 'Esse link encurtado já existe.';
      }

      setErrorCreatingNewLink({
        title: 'Erro no cadastro',
        description,
      });

      setOpenToastError(true);
    }
  };

  return (
    <>
      <form
        onSubmit={handleSubmit(handleSaveLink)}
        className="max-w-190 w-full md:min-w-145 flex flex-col flex-1 gap-10 md:gap-12 bg-gray-100 rounded-lg p-12 md:p-16"
      >
        <h2 className="text-lg text-gray-600">Novo link</h2>

        <div className="flex flex-col gap-8">
          <Input
            id="input-original-link"
            label="Link Original"
            placeholder="www.exemplo.com.br"
            error={errors.originalLink?.message}
            data-testid="input-original-link"
            {...register('originalLink')}
          />

          <Input
            id="input-shortened-link"
            label="Link Encurtado"
            fixedPlaceholder="brev.ly/"
            error={errors.shortenedLink?.message}
            data-testid="input-shortened-link"
            {...register('shortenedLink')}
          />
        </div>

        <Button
          type="submit"
          disabled={isSubmitting}
          data-testid="button-save-link"
        >
          {isSubmitting ? 'Salvando...' : 'Salvar link'}
        </Button>
      </form>

      <Toast.Provider swipeDirection="right">
        <Toast.Root
          open={openToastError}
          onOpenChange={setOpenToastError}
          duration={5000}
          className="w-150 flex flex-row gap-6 items-start justify-start bg-[#f1d4da] p-8 rounded-lg shadow-lg shadow-gray-300"
          data-testid="toast-creation-error"
        >
          <WarningCircle
            color="var(--color-danger)"
            size="1.25rem"
            weight="fill"
          />

          <div className="flex-1">
            <Toast.Title className="text-md text-danger">
              {errorCreatingNewLink.title}
            </Toast.Title>

            <Toast.Description className="text-sm text-danger mt-2">
              {errorCreatingNewLink.description}
            </Toast.Description>
          </div>

          <Toast.Action
            altText="Botão para fechar a caixa de erro"
            className="cursor-pointer text-danger"
          >
            <X />
          </Toast.Action>
        </Toast.Root>

        <Toast.Viewport className="fixed bottom-6 left-1/2 transform -translate-x-1/2 md:bottom-12 md:right-12 md:left-auto md:transform-none md:translate-x-0" />
      </Toast.Provider>
    </>
  );
};
