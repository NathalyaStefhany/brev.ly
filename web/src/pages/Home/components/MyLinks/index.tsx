import React from 'react';

import {
  DownloadSimple,
  LinkBreak,
  Link as LinkIcon,
} from '@phosphor-icons/react';
import { useInfiniteQuery } from '@tanstack/react-query';
import * as ScrollArea from '@radix-ui/react-scroll-area';
import { Button } from '@/components/Button';
import { api } from '@/service/api';
import { Spinner } from '@/components/Spinner';
import { Link } from '@/pages/Home/components/MyLinks/Link';
import { EmptyState } from '@/components/EmptyState';

const PAGE_SIZE = 20;

type ShortenedLinkInfo = {
  id: string;
  originalLink: string;
  shortenedLink: string;
  quantityAccesses: number;
  createdAt: string;
};

type ShortenedLinkList = {
  total: number;
  page: number;
  pageSize: number;
  data: ShortenedLinkInfo[];
};

export const MyLinks: React.FC = () => {
  const {
    data: listOfLinks,
    isError: listOfLinksIsError,
    isFetching: listOfLinksIsFetching,
    isRefetching: listOfLinksIsRefetching,
    hasNextPage,
    fetchNextPage,
  } = useInfiniteQuery<ShortenedLinkList>({
    queryKey: ['links list'],
    queryFn: async ({ pageParam }) => {
      const { data } = await api.get(
        `/shortened-links?page=${pageParam}&pageSize=${PAGE_SIZE}`,
      );

      return data;
    },
    initialPageParam: 1,
    getNextPageParam: (lastPage, allPages) => {
      if (lastPage.total > allPages.length * PAGE_SIZE) {
        return lastPage.page + 1;
      }

      return undefined;
    },
  });

  const onScroll = (event: React.UIEvent<HTMLDivElement>): void => {
    const { scrollHeight, scrollTop, clientHeight } = event.currentTarget;

    const distanceFromScrollToEnd = scrollHeight - scrollTop - clientHeight;
    const scrollIsMiddle = distanceFromScrollToEnd <= clientHeight / 2;

    if (scrollIsMiddle && hasNextPage && !listOfLinksIsFetching) {
      fetchNextPage();
    }
  };

  const quantityOfLinks = (listOfLinks && listOfLinks.pages[0].total) ?? 0;
  const listOfLinksIsEmpty = quantityOfLinks === 0;

  return (
    <div className="relative max-w-190 w-full md:max-w-290 md:min-w-190 h-fit flex flex-col flex-1 gap-8 md:gap-10 bg-gray-100 rounded-lg p-12 md:p-16">
      {listOfLinksIsRefetching && (
        <div className="absolute top-0 left-0 h-1 w-full animate-border bg-size-[100px_auto] md:bg-size-[200px_auto] bg-no-repeat bg-gradient-to-r from-blue-base to-blue-base" />
      )}

      <div className="flex flex-row items-center justify-between">
        <h2 className="text-lg text-gray-600">Meus links</h2>

        <Button variant="secondary" icon={DownloadSimple} disabled>
          Baixar CSV
        </Button>
      </div>

      <div className="border-t border-gray-200">
        {listOfLinksIsError ? (
          <EmptyState
            icon={<LinkBreak size="2rem" color="var(--color-danger)" />}
            description="Erro ao carregar os links cadastrados"
            data-testid="container-link-list-error"
          />
        ) : !listOfLinks ? (
          <EmptyState
            icon={<Spinner />}
            description="Carregando..."
            data-testid="container-link-list-loading"
          />
        ) : listOfLinksIsEmpty ? (
          <EmptyState
            icon={<LinkIcon size="2rem" color="var(--color-gray-400)" />}
            description="Ainda nÃ£o existem links cadastrados"
            data-testid="container-link-list-empty"
          />
        ) : (
          <ScrollArea.Root type="auto" className="w-full">
            <ScrollArea.Viewport
              onScroll={onScroll}
              className="overflow-hidden"
              style={{
                maxHeight: 'calc(100vh - 19rem)',
                minHeight: quantityOfLinks < 4 ? 'fit-content' : '14.125rem',
              }}
              data-testid="container-my-links-scroll"
            >
              <div className="md:w-full flex flex-col gap-3 md:gap-4">
                {listOfLinks.pages.map(({ page, data }) =>
                  data.map((link, index) => {
                    const isFirstLink = index === 0 && page === 1;

                    return (
                      <Link
                        key={link.id}
                        isFirstLink={isFirstLink}
                        info={link}
                      />
                    );
                  }),
                )}

                {hasNextPage && <Link isFirstLink={false} isLoading />}
              </div>
            </ScrollArea.Viewport>

            <ScrollArea.Scrollbar
              orientation="vertical"
              className="flex w-4 mr-[-0.875rem] bg-gray-200 rounded-full transition-colors duration-150 ease-out"
            >
              <ScrollArea.Thumb className="flex-1 rounded-full bg-blue-base hover:bg-blue-dark" />
            </ScrollArea.Scrollbar>
          </ScrollArea.Root>
        )}
      </div>
    </div>
  );
};
