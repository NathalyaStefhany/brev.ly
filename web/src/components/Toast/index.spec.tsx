import { Toast } from '@/components/Toast';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

describe('Toast tests', () => {
  it('should render correctly when type is error', () => {
    const { container } = render(
      <Toast
        id="toast"
        type="error"
        title="Toast title"
        description="Toast description"
        open
        onOpenChange={vi.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly when type is information', () => {
    const { container } = render(
      <Toast
        id="toast"
        type="information"
        title="Toast title"
        description="Toast description"
        open
        onOpenChange={vi.fn()}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
