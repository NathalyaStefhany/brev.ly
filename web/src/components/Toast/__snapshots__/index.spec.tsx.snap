// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast tests > should render correctly when type is error 1`] = `
<div>
  <div
    aria-label="Notifications (F8)"
    role="region"
    style=""
    tabindex="-1"
  >
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
    <ol
      class="z-3 fixed bottom-6 left-1/2 transform -translate-x-1/2 md:bottom-12 md:right-12 md:left-auto md:transform-none md:translate-x-0"
      tabindex="-1"
    >
      <li
        aria-atomic="true"
        aria-live="off"
        class="max-w-screen w-180 flex flex-row gap-6 items-start justify-start bg-[#f1d4da] p-8 rounded-lg shadow-lg shadow-gray-300"
        data-radix-collection-item=""
        data-state="open"
        data-swipe-direction="right"
        data-testid="toast"
        role="status"
        style="user-select: none;"
        tabindex="0"
      >
        <svg
          fill="var(--color-danger)"
          height="1.25rem"
          viewBox="0 0 256 256"
          width="1.25rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-8,56a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
          />
        </svg>
        <div
          class="flex-1"
        >
          <div
            class="text-md text-danger"
          >
            Toast title
          </div>
          <div
            class="text-sm text-danger mt-2"
          >
            Toast description
          </div>
        </div>
        <button
          class="cursor-pointer text-danger"
          data-radix-toast-announce-alt="Botão para fechar a caixa de aviso"
          data-radix-toast-announce-exclude=""
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
            />
          </svg>
        </button>
      </li>
    </ol>
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Toast tests > should render correctly when type is information 1`] = `
<div>
  <div
    aria-label="Notifications (F8)"
    role="region"
    style=""
    tabindex="-1"
  >
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
    <ol
      class="z-3 fixed bottom-6 left-1/2 transform -translate-x-1/2 md:bottom-12 md:right-12 md:left-auto md:transform-none md:translate-x-0"
      tabindex="-1"
    >
      <li
        aria-atomic="true"
        aria-live="off"
        class="max-w-screen w-180 flex flex-row gap-6 items-start justify-start bg-[#d6d8ef] p-8 rounded-lg shadow-lg shadow-gray-300"
        data-radix-collection-item=""
        data-state="open"
        data-swipe-direction="right"
        data-testid="toast"
        role="status"
        style="user-select: none;"
        tabindex="0"
      >
        <svg
          fill="var(--color-blue-base)"
          height="1.25rem"
          viewBox="0 0 256 256"
          width="1.25rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm-8,56a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm8,104a12,12,0,1,1,12-12A12,12,0,0,1,128,184Z"
          />
        </svg>
        <div
          class="flex-1"
        >
          <div
            class="text-md text-blue-base"
          >
            Toast title
          </div>
          <div
            class="text-sm text-blue-base mt-2"
          >
            Toast description
          </div>
        </div>
        <button
          class="cursor-pointer text-blue-base"
          data-radix-toast-announce-alt="Botão para fechar a caixa de aviso"
          data-radix-toast-announce-exclude=""
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
            />
          </svg>
        </button>
      </li>
    </ol>
    <span
      aria-hidden="true"
      style="position: fixed; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
      tabindex="0"
    />
  </div>
</div>
`;
